nixnet/database/_find_object.py

```diff
@@ -0,0 +1,52 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import typing  # NOQA: F401
+
+from nixnet import _cconsts
+from nixnet import _errors
+from nixnet import _funcs
+from nixnet import constants
+
+
+def find_object(parent_handle, object_class, object_name):
+    # type: (int, constants.ObjectClass, typing.Text) -> object
+    found_handle = _funcs.nxdb_find_object(parent_handle, object_class, object_name)
+
+    if found_handle == 0:
+        # A bit of an abuse of an error code
+        _errors.check_for_error(_cconsts.NX_ERR_DATABASE_OBJECT_NOT_FOUND)
+        # The above line will raise an exception.
+        # However, mypy doesn't know that and will complain if the following return is missing.
+        return None
+    elif object_class == constants.ObjectClass.CLUSTER:
+        from nixnet.database._cluster import Cluster
+        return Cluster(found_handle)
+    elif object_class == constants.ObjectClass.FRAME:
+        from nixnet.database._frame import Frame
+        return Frame(found_handle)
+    elif object_class == constants.ObjectClass.PDU:
+        from nixnet.database._pdu import Pdu
+        return Pdu(found_handle)
+    elif object_class == constants.ObjectClass.ECU:
+        from nixnet.database._ecu import Ecu
+        return Ecu(found_handle)
+    elif object_class == constants.ObjectClass.SIGNAL:
+        from nixnet.database._signal import Signal
+        return Signal(found_handle)
+    elif object_class == constants.ObjectClass.SUBFRAME:
+        from nixnet.database._subframe import SubFrame
+        return SubFrame(found_handle)
+    elif object_class == constants.ObjectClass.LIN_SCHED:
+        from nixnet.database._lin_sched import LinSched
+        return LinSched(found_handle)
+    elif object_class == constants.ObjectClass.LIN_SCHED_ENTRY:
+        from nixnet.database._lin_sched_entry import LinSchedEntry
+        return LinSchedEntry(found_handle)
+    else:
+        # A bit of an abuse of an error code
+        _errors.check_for_error(_cconsts.NX_ERR_INVALID_PROPERTY_VALUE)
+        # The above line will raise an exception.
+        # However, mypy doesn't know that and will complain if the following return is missing.
+        return None
```

Comment:
Looks like you could make the return type a union of the regular and [`NoReturn`](https://mypy.readthedocs.io/en/latest/kinds_of_types.html#the-noreturn-type)